src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java

```diff
@@ -143,6 +146,9 @@
 
     private AWSCredentials mCreds = null;
 
+    private Queue<Event> pendingIceCandidatesQueue = new LinkedList<>();
+    private HashMap<String, Boolean> peerConnectionFoundMap = new HashMap<String, Boolean>();
```

Comment:
Seems like you need a HashSet instead of a HashMap

Reply:
Or, perhaps you consider your data structures again. If multi-session is what we want to support then you could

1) Use a list of pending messages together with their clientid
2) Use a hash map of a client id -> linked list of pending messages

Reply:
I have updated the data structures. There are two maps now: 1 that maps client ID to ice candidate queue and the other that maps client ID to a boolean flag to update if peer connection for the client ID is established or not.

Reply:
Thanks. As we had a quick discussion, in this case in order to forward compat with multi-peer application (which it seems that we don't do currently) it makes more sense to have a hash map of string (client ID) -> peer connection object.

A presence of an entry in this map will mean that the peer connection object is created and the offer/answer has been received so the potential ICE candidate (in-order message) should go directly to the peer connection.